src/main/java/com/agileavengers/icuconnectbackend/service/implementation/UserService.java

```diff
@@ -69,26 +83,43 @@ public Set<CommunityDto> updateCommunityRelation(String username, String moduleI
 
     @Override
     public Set<CommunityDto> getJoinedCommunities(String username) {
-        Optional<User> optUser = userRepository.findByUsername(username);
-        if (optUser.isEmpty()) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User does not exist");
-        }
-        User user = optUser.get();
+        User user = getUserFromDb(username);
         return user.getSubscriptionSet().stream().map(communityMapper::toDto).collect(Collectors.toSet());
     }
 
     @Override
     public RatingDto getCommunityRating(String username, String moduleId) {
-        Optional<User> optUser = userRepository.findByUsername(username);
-        if (optUser.isEmpty()) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User does not exist");
-        }
-        User user = optUser.get();
+        User user = getUserFromDb(username);
         Optional<Community> optCommunity = communityRepository.findCommunityByModuleId(moduleId);
         if (optCommunity.isEmpty()) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Community does not exist");
         }
         Optional<Rating> rating = ratingRepository.findByCommunity_ModuleIdAndCreator_Id(moduleId, user.getId());
         return rating.map(ratingMapper::toDto).orElse(null);
     }
+
+
+    private User getUserFromDb(String username) {
+        Optional<User> optUser = userRepository.findByUsername(username);
+        if (optUser.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User does not exist");
+        }
+        return optUser.get();
+    }
+
+    private User updateFields(User user, UserDetailDto update) {
```

Comment:
As soon as the fields are changed and persisted we need to generate a new token for the user, otherwise they will have to log in again. Maybe return the token in the response header?

```
UserDetails userDetails = userDetailsService.loadUserByUsername(authenticationRequest.getUsername());
String token = jwtTokenUtil.generateToken(userDetails);
```

Reply:
Created a task #98